utilities/backupable/backupable_db_test.cc

```diff
@@ -1031,6 +1043,102 @@ TEST_P(BackupableDBTestWithParam, VerifyBackup) {
   CloseDBAndBackupEngine();
 }
 
+// This test verifies that the verifyBackup method correctly identifies
+// invalid backups
+TEST_P(BackupableDBTestWithParam, BlobFileDeletionAfterBackup) {
+  const int keys_iteration = 5000;
+  options_.enable_blob_files = true;
+
+  std::string shared_dir = backupdir_;
+  if (backupable_options_->share_files_with_checksum) {
+    shared_dir += "/shared_checksum";
+  } else {
+    shared_dir += "/shared";
+  }
+
+  OpenDBAndBackupEngine(true);
+  // create a backup
+  FillDB(db_.get(), 0, keys_iteration);
+  ASSERT_OK(backup_engine_->CreateNewBackup(db_.get(), true));
+
+  CloseDBAndBackupEngine();
+
+  OpenDBAndBackupEngine();
+
+  Random rnd(6);
+  std::vector<std::string> files;
+  std::vector<FileAttributes> children;
+  ASSERT_OK(file_manager_->GetChildrenFileAttributes(shared_dir, &children));
+
+  for (const auto& child : children) {
+    if (child.name.find(".blob") != std::string::npos &&
+        child.size_bytes != 0) {
+      files.push_back(child.name);
+    }
+  }
+
+  // ---------- case 1. - valid backup --------------------
+  ASSERT_TRUE(backup_engine_->VerifyBackup(1).ok());
+
+  // ---------- case 2. - delete a random blob file -------
+  {
+    size_t i = rnd.Uniform(static_cast<int>(files.size()));
+    ASSERT_OK(file_manager_->DeleteFile(shared_dir + "/" + files[i]));
+    ASSERT_TRUE(backup_engine_->VerifyBackup(1).IsNotFound());
+  }
+
```

Comment:
I am trying to corrupt the blob file in shared dir. Once the file is corrupted, deleting the blob  file gives corruption error because of shared dir. That's why I added two separate test.